src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp

```diff
@@ -772,35 +772,53 @@ void InterpreterMacroAssembler::merge_frames(Register Rsender_sp, Register retur
 
 void InterpreterMacroAssembler::narrow(Register result) {
   Register ret_type = R5_scratch1;
-  ld_PPC(R5_scratch1, in_bytes(Method::const_offset()), R27_method);
-  lbz_PPC(ret_type, in_bytes(ConstMethod::result_type_offset()), R5_scratch1);
+  Register scratch = R6_scratch2;
+  ld(R5_scratch1, R27_method, in_bytes(Method::const_offset()));
+  lbu(ret_type, R5_scratch1, in_bytes(ConstMethod::result_type_offset()));
 
-  Label notBool, notByte, notChar, done;
+  Label notBool, notByte, notChar, byteOne, shortOne, done;
 
   // common case first
-  cmpwi_PPC(CCR0, ret_type, T_INT);
-  beq_PPC(CCR0, done);
+  addi(scratch, R0, T_INT);
+  beq(scratch, ret_type, done);
 
-  cmpwi_PPC(CCR0, ret_type, T_BOOLEAN);
-  bne_PPC(CCR0, notBool);
+  addi(scratch, R0, T_BOOLEAN);
+  bne(scratch, ret_type, notBool);
   andi(result, result, 0x1);
-  b_PPC(done);
+  j(done);
 
   bind(notBool);
-  cmpwi_PPC(CCR0, ret_type, T_BYTE);
-  bne_PPC(CCR0, notByte);
-  extsb_PPC(result, result);
-  b_PPC(done);
+  addi(scratch, R0, T_BYTE);
+  bne(scratch, ret_type, notByte);
+  // sign-extend lower 8 bits
+  addi(scratch, R0, 0x80);
+  bge(result, scratch, byteOne);
+  andi(result, result, 0xff);
+  j(done);
+  bind(byteOne);
+  ori(result, result, 0xffffff00);
+  j(done);
 
   bind(notByte);
-  cmpwi_PPC(CCR0, ret_type, T_CHAR);
-  bne_PPC(CCR0, notChar);
-  andi(result, result, 0xffff);
-  b_PPC(done);
+  addi(scratch, R0, T_CHAR);
+  bne(scratch, ret_type, notChar);
+  addi(scratch, R0, 0xfff);
+  lui(scratch, 0xf);
```

Comment:
I would propose use `load_const` or similar, becasue it's potentially more efficient and it should have fewer bugs
* it should be `lui` + `addi`, as lui clears low bits
* 0xfff will be sign extended, so you need to load a (value+1) with lui